### PR TITLE
add option to manually input forcal length

### PR DESCRIPTION
**Pull request by floriantschopp**

### CODE REVIEW

File: aslam_cv/aslam_cameras/include/aslam/cameras/implementation/PinholeProjection.hpp [hunk 1]
```diff
@@ -777,9 +778,14 @@ bool PinholeProjection<DISTORTION_T>::initializeIntrinsics(const std::vector<Gri
...
-  //get the median of the guesses
-  if(f_guesses.empty())
-    return false;
+  // Get the median of the guesses if available.
+  if(f_guesses.empty()) {
+   float input_guess;
+   std::cout << "Initialization of focal length failed. Provide manual initialization: ";
+   std::cin >> input_guess;
+   std::cout << "Initialize focal length to " << input_guess << "\n";
+   f_guesses.push_back(static_cast<float>(input_guess));
```
Comment (LBern) [inline aslam_cv/aslam_cameras/include/aslam/cameras/implementation/PinholeProjection.hpp]: `emplace_back`?

File: aslam_cv/aslam_cameras/include/aslam/cameras/implementation/PinholeProjection.hpp [hunk 2]
```diff
@@ -777,9 +778,14 @@ bool PinholeProjection<DISTORTION_T>::initializeIntrinsics(const std::vector<Gri
...
 
-  //get the median of the guesses
-  if(f_guesses.empty())
-    return false;
+  // Get the median of the guesses if available.
+  if(f_guesses.empty()) {
+   float input_guess;
+   std::cout << "Initialization of focal length failed. Provide manual initialization: ";
+   std::cin >> input_guess;
+   std::cout << "Initialize focal length to " << input_guess << "\n";
```
Comment (LBern) [inline aslam_cv/aslam_cameras/include/aslam/cameras/implementation/PinholeProjection.hpp]: Initializing?

File: aslam_cv/aslam_cameras/include/aslam/cameras/implementation/PinholeProjection.hpp [hunk 3]
```diff
@@ -777,9 +778,14 @@ bool PinholeProjection<DISTORTION_T>::initializeIntrinsics(const std::vector<Gri
...
   }
 
-  //get the median of the guesses
-  if(f_guesses.empty())
-    return false;
+  // Get the median of the guesses if available.
+  if(f_guesses.empty()) {
+   float input_guess;
+   std::cout << "Initialization of focal length failed. Provide manual initialization: ";
+   std::cin >> input_guess;
```
Comment (LBern) [inline aslam_cv/aslam_cameras/include/aslam/cameras/implementation/PinholeProjection.hpp]: shall we check the input, e.g. positive focal lengths?

File: aslam_cv/aslam_cameras/include/aslam/cameras/implementation/PinholeProjection.hpp (hunk 1 above)
Comment (floriantschopp) [reply aslam_cv/aslam_cameras/include/aslam/cameras/implementation/PinholeProjection.hpp]: Trying to stay conistent with https://github.com/ethz-asl/kalibr/blob/a5186202b7bf1db7d9e0828ef24dff2bba43340d/aslam_cv/aslam_cameras/include/aslam/cameras/implementation/PinholeProjection.hpp#L776

**Pull request merged**